beluga/include/beluga/sensor/data/regular_grid.hpp

```diff
@@ -74,8 +74,14 @@ class BaseRegularGrid2 : public ciabatta::ciabatta_top<Derived> {
    * \return Grid cell coordinates.
    */
   [[nodiscard]] Eigen::Vector2i cell_near(double x, double y) const {
-    const auto xi = static_cast<int>(std::floor(x / this->self().resolution()));
-    const auto yi = static_cast<int>(std::floor(y / this->self().resolution()));
+    const auto inv_resolution = 1. / this->self().resolution();
+    const auto scaled_x = x * inv_resolution;
+    const auto scaled_y = y * inv_resolution;
+    // Truncate towards zero. Use a poor man's replacement of
+    // `std::floor` because the later is awfully slow and this
+    // function is performance critical.
```

Comment:
This is not truncating towards zero, but instead returning the greatest integer not greater than scaled_x/scaled_y (which for positive numbers is the same)

```suggestion
    // Use a poor man's replacement of `std::floor` because
    // the later is awfully slow and this function is performance
    // critical.
```